ci: failure to download 0.14.3-win64.zip in Win test cross-built

This is happening semi-frequently:

https://github.com/bitcoin/bitcoin/actions/runs/19520873587/job/55885060054:
```bash
Run ./test/get_previous_releases.py --target-dir $PREVIOUS_RELEASES_DIR

Download failed: Remote end closed connection without response
Releases directory: D:\a\_temp/previous_releases
Fetching: https://bitcoincore.org/bin/bitcoin-core-0.14.3/bitcoin-0.14.3-win64.zip
Error: Process completed with exit code 1.
```

https://github.com/bitcoin/bitcoin/actions/runs/19239483669/job/55006759227:
```bash
 Releases directory: D:\a\_temp/previous_releases
Fetching: https://bitcoincore.org/bin/bitcoin-core-0.14.3/bitcoin-0.14.3-win64.zip

Download failed: [WinError 10054] An existing connection was forcibly closed by the remote host
Downloading: [----------------------------------------] 0.0%
Error: Process completed with exit code 1.
```

https://github.com/bitcoin/bitcoin/actions/runs/19481968460/job/55756476621:
```bash

Run ./test/get_previous_releases.py --target-dir $PREVIOUS_RELEASES_DIR

Download failed: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
Releases directory: D:\a\_temp/previous_releases
Fetching: https://bitcoincore.org/bin/bitcoin-core-0.14.3/bitcoin-0.14.3-win64.zip
Error: Process completed with exit code 1.
```

--- a/test/get_previous_releases.py
+++ b/test/get_previous_releases.py
@@ -168,7 +168,13 @@ def download_binary(tag, args) -> int:
         download_from_url(archive_url, archive)
     except Exception as e:
         print(f"\nDownload failed: {e}", file=sys.stderr)
-        return 1
+        print("Retrying download after failure ...", file=sys.stderr)
+        time.sleep(12)
+        try:
+            download_from_url(archive_url, archive)
+        except Exception as e2:
+            print(f"\nDownload failed a second time: {e2}", file=sys.stderr)
+            return 1
 
     hasher = hashlib.sha256()
     with open(archive, "rb") as afile:

